__tests__/app.test.js

```diff
@@ -29,10 +31,83 @@ describe("GET /api/topics", () => {
             })
           );
           console.log("=================>", topic);
-        }); //in the for each test for slug and description - test for data type not ACTUAL data
+        });
       });
   });
 });
+//ticket #14 connecting to article:id
+describe("GET /api/articles/:article_id", () => {
+  test("status: 200 - connecting to article endpoint", () => {
+    return request(app)
+      .get("/api/articles/4")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toEqual({
+          article_id: 4,
+          title: "Student SUES Mitch!",
+          topic: "mitch",
+          author: "rogersop",
+          body: "We all love Mitch and his wonderful, unique typing style. However, the volume of his typing has ALLEGEDLY burst another students eardrums, and they are now suing for damages",
+          created_at: expect.any(String),
+          votes: 0,
+        });
+      });
+  });
+  //unhappy path below
+  test("error: 404 - article not found", () => {
+    return request(app)
+      .get("/api/articles/9000")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("article not found");
+      });
+  });
+
+  test("status:400 - responds with an error message when passed a bad request", () => {
+    return request(app)
+      .get("/api/articles/potato")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
+  });
+
+  test("status:404 - responds with an error message when passed a bad request", () => {
+    return request(app)
+      .get("/badroute")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("path not found");
+      });
+  });
+});
+
+//in the for each test for slug and description - test for data type not ACTUAL data
+
 //check out express syntax
 //now use express to check if this endpoint is giving back code 200
 //and returning back the slug and description.
+
+//THIS CODE WILL BE USEFUL FOR LATER ON TICKET #9
+// describe.only("GET /api/articles/:article_id", () => {
+//   test("status: 200 - connecting to article endpoint", () => {
+//     return request(app)
+//       .get("/api/articles/4")
+//       .expect(200)
+//       .then(({ body }) => {
+//         expect(body).toHaveLength(12);
+//         body.forEach((article) => {
+//           expect(article).toEqual(
+//             expect.objectContaining({
+//               title: expect.any(String),
+//               topic: expect.any(String),
+//               author: expect.any(String),
+//               body: expect.any(String),
+//               created_at: expect.any(String),
+//               votes: expect.any(Number),
+//             })
+//           );
+//           // console.log("=================>", article);

```

Comment:
where has this code come from? if you have not worked on ticket 9, why have you got code which would be useful for it?